pymeasure/instruments/instrument.py

```diff
@@ -318,7 +318,7 @@ class FakeInstrument(Instrument):
 
     def __init__(self, adapter=None, name=None, includeSCPI=False, **kwargs):
         super().__init__(
-            FakeAdapter(),
+            adapter or FakeAdapter(),
```

Comment:
Please revert this - otherwise if you pass a string as `adapter`, the Instrument init will go the usual route and try to make a VisaAdapter which will fail. This is why the doctests fail.

Reply:
I can't just revert. The problem is that to test `preprocess_reply` at the adapter level one should be able to pass kwargs to the Adapter constructor. I understand you want to have your strings in the documentation but still get a `FakeAdapter`, right?

Reply:
a90e4bd should fix this. Let me know if this is ok?

Reply:
I think the FakeAdapter needs an init that is adapted to the new structure, e.g. try this:
```
    def __init__(self, preprocess_reply=None, **kwargs):
        self.preprocess_reply = preprocess_reply
        # kwargs are discarded as they are non-functional.
```
then here you should be able to say 
```
        super().__init__(
            FakeAdapter(**kwargs),
```

Reply:
^ to be honest, I would prefer the outlined approach (if it proves to work as intended) to a90e4bd .

Reply:
`FakeAdapter` has this init already! inherited from `Adapter`. I would not like your proposal because you mix `kwargs` of `Adapter` and `Instrument`, but I guess for pyvisa instruments that is what you already do... If you dislike my fix let me know

Reply:
sorry, again crossed comments. I now implemented your suggestion. Seems to work as intended.